go/vt/vtgate/executor_select_test.go

```diff
@@ -3011,6 +3011,8 @@ func TestStreamOrderByLimitWithMultipleResults(t *testing.T) {
 	}
 
 	executor := NewExecutor(context.Background(), serv, cell, resolver, true, false, testBufferSize, cache.DefaultConfig, nil, false, querypb.ExecuteOptions_Gen4)
+	// some sleep for all goroutines to start
+	time.Sleep(100 * time.Millisecond)
 	before := runtime.NumGoroutine()
 
```

Comment:
nit: cannot rely on this, can still keep the test flaky.

Reply:
Yes, we should not count on anything happening during a wall clock period of time in the CI. We regularly get no CPU time for multiple seconds.

Reply:
I will let the query serving folk fix this, because I am not sure how to use a different method. Currently It uses the same wait used below in this test for a similar purpose. Maybe keep getting the routine count and stop when it doesn't change within some interval?